### PR TITLE
[CI] Enable unit tests

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes: RNBW-4791

## What changed (plus any additional context for devs)

**Reason for PR:**
- Unit tests were disabled here by Skylar (RIP): https://github.com/rainbow-me/rainbow/pull/5400/files

**Reenabling on CI:**
- `yarn test`

**Added:**
- New Unit test covering `convertAmountToNativeDisplayWorklet`
- Simple workflow. maybe want dev eyes to make sure test cases are extensive and outcomes are expected

Changes of note:
- Changed where `'ts-jest': { tsconfig: 'tsconfig.jest.json',},` is located in the config to remove a deprication warning
- Added `src/__swaps__`...`findNiceIncrement...` to the jest ignore. Currently doesn't work bc reanimated breaks our tests.
- **Skipping** analytics test 
- **Skipping** favorite dapps
- **Fixed** logger test
- **Fixed** app sessions test

Follow up:
- mock reanimated to fix `__swaps__` tests
- fix analytics test
- fix favorite dapps test

### CODE REVIEW

**Review by brunobar79 — APPROVED** [review]
LGTM!

File: jest.config.js [hunk 1]
```diff
@@ -5,20 +5,23 @@ const { compilerOptions } = require('./tsconfig');
...
   preset: 'react-native',
   setupFiles: ['./config/test/jest-setup.js'],
-  testPathIgnorePatterns: ['node_modules', 'e2e'],
-  globals: {
-    'ts-jest': {
-      tsconfig: 'tsconfig.jest.json',
-    },
-  },
+  // need to mock reanimated and some other things to unblock NiceIncrementerFormatter.test.ts
+  testPathIgnorePatterns: ['node_modules', 'e2e', 'src/__swaps__/screens/Swap/__tests__/NiceIncrementerFormatter.test.ts'],
```
[the remark this answers is not in the document]
Comment (BrodyHughes) [reply jest.config.js:9]: will do!

[the remark this answers is not in the document]
Comment (BrodyHughes) [reply jest.config.js]: updated jest regex + renamed the test

**Review by benisgold — APPROVED** [review]

**Pull request merged**